app/controllers/front_controller.rb

```diff
@@ -0,0 +1,79 @@
+class FrontController < ApplicationController
+  before_filter :fetch_requested_content
+
+  rescue_from RecordNotFound, with: :cacheable_404
+
+  def call
+    # delegate this request to the appropriate controller
+    format_controller(@format).public_send(
+      action_from_request_method(request.method)
+    ) and return
+  end
+
+  private
+
+  def fetch_requested_format
+    # try to fetch the requested content from content-store
+    if version_from_content_store(params[:slug])
+
+      # memoize the content to the request object
+      # to prevent a second call to the content-store
+      request.version = @version
+      @format = @version.format
+
+    # failing that, fallback to the content-api
+    elsif edition_from_content_api(params[:slug], params[:edition])
+      request.edition = @edition
+      @format = @edition.format
+
+    # it doesn't exist
+    else
+      raise ActiveRecord::RecordNotFound
+    end
+  end
+
+  # fetches requested content from content-api
+  def edition_from_content_api(slug, edition_id)
+    @edition ||= fetch_artefact(slug, edition_id, nil)
+  end
+
+  # fetches requested content from content-store
+  def version_from_content_store(slug)
+    @version ||= content_store.content_item("/#{slug}")
+  end
+
+  # longhand mapping of format to controller
+  # no magic here.
+  def format_controller(format)
+    case format
+    when 'help'
+      HelpController.new
+    when 'guide'
+      GuideController.new
+    when 'licence'
+      LicenceController.new
+    else
+      # legacy
+      RootController.new
+    end.tap do |controller|
+      controller.request = request
+      controller.response = response
+    end
+  end
+
+  # https://gdstechnology.blog.gov.uk/2017/06/01/how-we-implemented-the-front-controller-pattern-in-rails-and-why-you-shouldnt-do-that
+  def action_from_method(method)
```

Comment:
I think we only need to support GET and POST for frontend, not necessarily important at this point though...